pkg/reconciler/common/ha.go

```diff
@@ -50,7 +62,7 @@ func HighAvailabilityTransform(obj base.KComponent, log *zap.SugaredLogger) mf.T
 		replicas := int64(*ha.Replicas)
 
 		// Transform deployments that support HA.
-		if u.GetKind() == "Deployment" && !haUnSupported(obj).Has(u.GetName()) {
+		if u.GetKind() == "Deployment" && !haUnSupported(obj).Has(u.GetName()) && !hasHorizontalPodAutoscaler(obj).Has(u.GetName()) {
```

Comment:
Should we also respect the user setting by setting replicas as usual and also by `minReplicas` in the horizontalPodAutoscaler, an example is here?

Reply:
Hmm.... I think https://github.com/openshift-knative/serverless-operator/blob/13833dae7e3f58294705cf2ad20ea6ae1573fccb/knative-operator/pkg/controller/knativekafka/knativekafka_controller.go#L659 also has the same issue with https://github.com/knative/operator/issues/1200 .

The operator always sets the replicas based on `Spec.HighAvailability.Replicas` to Deployment so it conflicts when HPA updates the deployment during the high load.

Reply:
To clarify, the issue happens the second case below.
#### Usual (no confliction happens :angel: ):
- operator sets replicas to `2` in deployment (based on `spec.HighAvailability.Replicas`)
- HPA sets replicas to `2` in deployment (based on `minReplicas`)

#### High load (confliction happens :fire:)
- operator sets replicas to `2` in deployment (based on `spec.HighAvailability.Replicas`)
- HPA sets replicas to `5` in deployment (based on the high load)

So, this PR changes to let operator skip changing the replicas for deployment which has HPA.

Reply:
@pierDipi ^^^ fyi.
Ok. So when operator deployment informer triggers a reconciliation we should skip applying setting replicas favoring hpa for the deployments that have an hpa config. 
Should we make sure though that the minScale for hpa is set to spec.HighAvailability.Replicas in all scenarios?

Reply:
I think current logic for hpa is fine.